ENIGMAsystem/SHELL/Universal_System/fileio.cpp

```diff
@@ -17,263 +17,214 @@
 
 // Simple, intuitive, integer based file I/O
 
-#include <cstdio> //fstream can get staked
-#include <string> //We will use string, though
-using namespace std;
-
-#include "darray.h" //Simpler vector with logarithmic time
-#include "fileio.h" //Simpler vector with logarithmic time
-
+#include "fileio.h"
+#include "Resources/AssetArray.h"
 #include "Widget_Systems/widgets_mandatory.h"
 
-// Implement a storage class for file information
-
-// We're going to set a standard at this point. On file close, the pointer shall be zeroed,
-// and the array shall be traversed from end to start in search of the lowest legal high ID
-// if the file's ID matches the high ID.
-
-namespace enigma
-{
-  struct openFile
-  {
-    FILE *f;      //FILE we opened, or NULL if it has been closed.
-    string sdata; //Use varies depending on type.
-    unsigned spos;      //position in sdata string
-    bool eof;
-
-    openFile(): f(NULL), sdata(), spos(0), eof(false) {};
-    openFile(FILE* a,string b): f(a), sdata(b), spos(0), eof(false) {};
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <iomanip>
+
+#ifdef DEBUG_MODE
+#define try_io_and_print(f) print_and_clear_fs_status(__FUNCTION__, f);
+#else
+#define try_io_and_print(f)
+#endif
+
+namespace enigma {
+  
+  struct file {
+    file() {}
+    file(const std::string& fName, std::ios_base::openmode mode) : fn(fName) { fs.open(fName, mode); }
+    file(file&& other) : fn(other.fn) { fs.swap(other.fs); }
+    file(file& other) : fn(other.fn) { fs.swap(other.fs); } 
```

Comment:
@RobertBColton josh says this copy constructor is wrong / unneeded but ass array won't work without it. Pls fix

Reply:
Was a bug 26b6cf6cfb34542b29e1a97985ba771bd0157dc8 in OffsetVector apparently.